src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleRequestSender.cs

```diff
@@ -1436,16 +1438,46 @@ private void AttachDebuggerToProcess(ITestHostLauncher customHostLauncher, Messa
 
         try
         {
-            var pid = _dataSerializer.DeserializePayload<int>(message);
+            // Handle EditorAttachDebugger2.
+            if (message.MessageType == MessageType.EditorAttachDebugger2)
```

Comment:
Can we employ a Builder pattern here ? Seeing the code is mostly identical for the two message types but the only relevant difference is the way we build the `AttachDebuggerInfo` object, it looks like we can avoid duplicating code.